src/Annotations/AbstractAnnotation.php

```diff
@@ -345,8 +345,17 @@ public function jsonSerialize()
 
         // $ref
         if (isset($data->ref)) {
-            // OAS 3.0 does not allow $ref to have siblings: http://spec.openapis.org/oas/v3.0.3#fixed-fields-18
-            $data = (object) ['$ref' => $data->ref];
+            // Only specific https://github.com/OAI/OpenAPI-Specification/blob/3.1.0/versions/3.1.0.md#reference-object
+            $ref = ['$ref' => $data->ref];
+            $defaultValues = get_class_vars(get_class($this));
+            foreach (['summary', 'description'] as $prop) {
+                if (property_exists($this, $prop)) {
+                    if ($this->$prop !== $defaultValues[$prop]) {
```

Comment:
Just a question: Is there any value in looking up default values this convoluted way? After all we know that `Generator::UNDEFINED`is the default value...